examples/runners/ash/src/main.rs

```diff
@@ -1,259 +1,229 @@
 
     pub pdevice: vk::PhysicalDevice,
-    pub device_memory_properties: vk::PhysicalDeviceMemoryProperties,
     pub queue_family_index: u32,
     pub present_queue: vk::Queue,
 
     pub surface: vk::SurfaceKHR,
+    pub surface_loader: khr::Surface,
     pub surface_format: vk::SurfaceFormatKHR,
-    pub surface_resolution: vk::Extent2D,
-
-    pub swapchain: vk::SwapchainKHR,
-    pub present_images: Vec<vk::Image>,
-    pub present_image_views: Vec<vk::ImageView>,
-
-    pub pool: vk::CommandPool,
-    pub draw_command_buffer: vk::CommandBuffer,
-    pub setup_command_buffer: vk::CommandBuffer,
-
-    pub present_complete_semaphore: vk::Semaphore,
-    pub rendering_complete_semaphore: vk::Semaphore,
-
-    pub draw_commands_reuse_fence: vk::Fence,
-    pub setup_commands_reuse_fence: vk::Fence,
 }
 
-impl ExampleBase {
-    pub fn render_loop<T, F: Fn(&Self) + 'static>(
-        self,
-        events_loop: winit::event_loop::EventLoop<T>,
-        f: F,
-    ) -> ! {
-        use winit::event::*;
-        use winit::event_loop::*;
-        events_loop.run(move |event, _window_target, control_flow| match event {
-            Event::RedrawEventsCleared { .. } => {
-                f(&self);
-            }
-            Event::WindowEvent { event, .. } => match event {
-                WindowEvent::KeyboardInput { input, .. } => {
-                    if let Some(VirtualKeyCode::Escape) = input.virtual_keycode {
-                        *control_flow = ControlFlow::Exit
-                    } else {
-                        *control_flow = ControlFlow::Wait
-                    }
-                }
-                WindowEvent::CloseRequested => *control_flow = ControlFlow::Exit,
-                _ => *control_flow = ControlFlow::Wait,
-            },
-            _ => *control_flow = ControlFlow::Wait,
-        });
-    }
-
-    pub fn new(
-        window_width: u32,
-        window_height: u32,
-        options: &Options,
-    ) -> (Self, winit::event_loop::EventLoop<()>) {
-        unsafe {
-            let events_loop = winit::event_loop::EventLoop::new();
-            let window = winit::window::WindowBuilder::new()
-                .with_title("Rust GPU - ash")
-                .with_inner_size(winit::dpi::LogicalSize::new(
-                    f64::from(window_width),
-                    f64::from(window_height),
-                ))
-                .build(&events_loop)
-                .unwrap();
-
-            cfg_if::cfg_if! {
-                if #[cfg(target_os = "macos")] {
-                    let entry = ash_molten::MoltenEntry::load().unwrap();
-                } else {
-                    let entry = ash::Entry::new().unwrap();
-                }
+impl RenderBase {
+    pub fn new(window: winit::window::Window, options: &Options) -> Self {
+        cfg_if::cfg_if! {
+            if #[cfg(target_os = "macos")] {
+                let entry = ash_molten::MoltenEntry::load().unwrap();
```

Comment:
It would be nice if somebody could run and test this on macOS before we merge it, mostly would like to see the resize code get tested there.

Reply:
Linux should get checked again as well.